docs/DeveloperGuide.adoc

```diff
@@ -240,43 +243,45 @@ image::CommitActivityDiagram.png[]
 
 * **Choice 1: (current choice)** Implements undo and redo of each Command in a separate Event object stored in the EventHistory
 ** Pros:
-*** Will use less memory as each frame is represented by an Event object instead of the entire model which consists of
-Exercise books and Regime books.
+*** Uses less memory to store Event objects as compared to entire copes of the Model.
 *** Open for extensions and close to modifications as the Event interface only contains undo and redo methods, and can
 be easily implemented when new Undoable commands are introduced.
 *** Obeys Single Responsibility Principle as Commands simply have to execute what it does and does not need to know
 how it is undone or redone
 ** Cons:
-*** Have to store the payload of Command objects to be accessed by Event objects
+*** UndoableCommand objects are forced to depend on EventPayloads when it does not actually use it directly.
 (e.g. for `delete`, the exercise being deleted).
 
 * **Choice 2:** Individual command knows how to undo/redo by itself.
 ** Pros:
-*** Will use less memory
+*** Uses less memory
```

Comment:
this point is actually pretty debatable. Because your commands still have to store the objects to undo.